File: testdata/swift/simple.swift
import Foundation

protocol Greeter {
    func greet(name: String) -> String
}

class SimpleGreeter: Greeter {
    func greet(name: String) -> String {
        return "Hello, \(name)!"
    }

    private func formatName(_ name: String) -> String {
        return name.capitalized
    }
}

struct AppConfig {
    let defaultName: String = "World"
}

enum Priority {
    case low
    case medium
    case high
}

func main() {
    let greeter = SimpleGreeter()
    let config = AppConfig()
    let result = greeter.greet(name: config.defaultName)
    print(result)
}

File: testdata/dart/simple.dart
import 'package:flutter/material.dart';
import 'helper.dart';

abstract class Greeter {
  String greet(String name);
}

mixin Loggable {
  void log(String message) {
    print(message);
  }
}

class SimpleGreeter extends Greeter with Loggable {
  @override
  String greet(String name) {
    log('Greeting $name');
    return 'Hello, $name!';
  }

  String _formatName(String name) {
    return name.toUpperCase();
  }
}

enum Priority { low, medium, high }

void main() {
  final greeter = SimpleGreeter();
  final result = greeter.greet('World');
  print(result);
  assist();
}
